src/evented-tokenizer.ts

```diff
@@ -1,23 +1,32 @@
 import { preprocessInput, isAlpha, isSpace } from './utils';
 
-function EventedTokenizer(delegate, entityParser) {
-  this.delegate = delegate;
-  this.entityParser = entityParser;
-
-  this.state = null;
-  this.input = null;
-
-  this.index = -1;
-  this.line = -1;
-  this.column = -1;
-  this.tagLine = -1;
-  this.tagColumn = -1;
-
-  this.reset();
-}
+export default class EventedTokenizer {
+  private delegate: any;
+  private entityParser: any;
+  private state: any;
+  private input: any;
+  private index: number;
+  private tagLine: number;
+  private tagColumn: number;
+
+  public line: number;
+  public column: number;
+
+  constructor(delegate, entityParser) {
```

Comment:
`constructor(private: delegate, private: entityParser) {}` would be all that is required if the other private state is initialized where it is declared